yclienttools/reportdataduplication.py

```diff
@@ -0,0 +1,222 @@
+    parser.add_argument("--yoda-version", type=str, default=None,
+                        help="Optional Yoda version (default defined in config)")
+    return parser.parse_args()
+
+
+def _get_output_columns(args: argparse.Namespace) -> List[str]:
+    """Determines output CSV column headers based on the presence of the environment flag."""
+    columns = ["Research Group", "Research Path", "Vault Path"]
+    return columns if args.environment is None else ["Environment"] + columns
+
+
+def _get_research_group_name(vault_group: str) -> str:
+    """Converts a vault group name to its corresponding research group name."""
+    return vault_group.replace("vault", "research")
+
+
+def _get_research_group_path(zone: str, research_group: str) -> str:
+    """Construct the full iRODS path for a research group."""
+    return f"/{zone}/home/{research_group}"
+
+
+def _get_collection_name(data_package_name: str) -> str:
+    """Get the base collection name from a data package name."""
+    return data_package_name.split("[")[0]
+
+
+def _get_research_matches(session: iRODSSession, research_path: str, collection_name: str) -> List[str]:
+    """
+    Get research collections matching the given collection name under the research group.
+    Includes:
+    - Direct match: if research_path ends in the collection_name
+    - Direct child: /research_path/collection_name
+    - Nested subcollections ending in /collection_name
+    """
+    matches = set()
+
+    # Match top-level
+    if research_path.split('/')[-1] == collection_name:
+        matches.add(research_path)
+
+    # Match subcollections
+    query = session.query(Collection.name).filter(
+        Like(Collection.name, f"{research_path}/%"),
+        Like(Collection.name, f"%/{collection_name}")
+    )
+    for row in query.get_results():
+        matches.add(row[Collection.name])
+
+    return sorted(matches)
+
+
+def _get_collection_tree(session: iRODSSession, coll_path: str) -> Dict[str, Dict]:
+    """
+    Return a dict containing:
+    - 'files': {relative_path: size}
+    - 'subcollections': sorted list of relative paths of all subcollections
+    """
+    result = {
+        "files": {},
+        "subcollections": set()
+    }
+
+    # Include base collection if it exists
+    if collection_exists(session, coll_path):
+        result["subcollections"].add("")  # base collection
+
+    # Get all subcollections (even empty ones)
+    query_subcolls = session.query(Collection.name).filter(
+        Like(Collection.name, f"{coll_path}/%")
+    )
+    for row in query_subcolls.get_results():
+        rel_coll_path = row[Collection.name].replace(coll_path, "").lstrip("/")
+        if rel_coll_path:
+            result["subcollections"].add(rel_coll_path)
+
+    # Get all files
+    query_files = session.query(Collection.name, DataObject.name, DataObject.size).filter(
+        Like(Collection.name, f"{coll_path}/%")
```

Comment:
This query returns only data objects in subcollections of `coll_path`. Can you add an additional query to also consider the data objects in `coll_path` itself?